### PR TITLE
Feature/UAR 954 Display Corporate MO contact email address

### PR DESCRIPTION
**Pull request by bmcgreevyCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-954

### Change description
Display Corporate MO contact email address

### CODE REVIEW

File: src/utils/update/managing.officer.mapper.ts [hunk 1]
```diff
@@ -88,3 +88,11 @@ export const mapMoPrivateAddress = (moPrivateData: ManagingOfficerPrivateData[],
     }
   }
 };
+
+export const mapMoPrivateEmailAddress = (moPrivateData: ManagingOfficerPrivateData[], ch_reference: string) => {
+  for (const managingOfficerData of moPrivateData) {
+    if (managingOfficerData.hashedId === ch_reference) {
+      return managingOfficerData.contactEmailAddress ?? undefined;
```
Comment (AndreKramer7654) [inline src/utils/update/managing.officer.mapper.ts:95]: you should not need the "?? undefined"

File: src/utils/update/managing.officer.mapper.ts [hunk 2]
```diff
@@ -88,3 +88,11 @@ export const mapMoPrivateAddress = (moPrivateData: ManagingOfficerPrivateData[],
     }
   }
 };
+
+export const mapMoPrivateEmailAddress = (moPrivateData: ManagingOfficerPrivateData[], ch_reference: string) => {
```
Comment (n-r-b) [inline src/utils/update/managing.officer.mapper.ts:92]: Same comment as in your other PR, may be good to combine the MO data mapping into one function (where we have duplicate code)

File: test/utils/update/fetch.managing.officers.private.data.spec.ts [hunk 3]
```diff
@@ -32,22 +32,37 @@ describe("fetchManagingOfficersPrivateData", () => {
...
     const usual_residential_address = appData.update?.review_managing_officers_individual?.[0].usual_residential_address;
+    const dob = appData.update?.review_managing_officers_individual?.[0].date_of_birth;
     expect(usual_residential_address?.property_name_number).toEqual("private_premises");
     expect(usual_residential_address?.line_1).toEqual("private_addressLine1");
     expect(usual_residential_address?.line_2).toEqual("private_addressLine2");
     expect(usual_residential_address?.town).toEqual("private_locality");
     expect(usual_residential_address?.county).toEqual("private_region");
     expect(usual_residential_address?.country).toEqual("private_country");
     expect(usual_residential_address?.postcode).toEqual("private_postalCode");
+    expect(usual_residential_address?.postcode).toEqual("private_postalCode");
```
Comment (AndreKramer7654) [inline test/utils/update/fetch.managing.officers.private.data.spec.ts:43]: line duplicated?

File: test/utils/update/fetch.managing.officers.private.data.spec.ts [hunk 4]
```diff
@@ -103,7 +118,7 @@ describe("fetchManagingOfficersPrivateData", () => {
     expect(appData.update?.review_managing_officers_corporate?.length).toBeUndefined();
   });
 
-  test("should not map MO private data if existing MO does not have ch_reference", async () => {
+  test("should not map Individual MO private data if existing MO does not have ch_reference", async () => {
     if (appData.update && appData.update.review_managing_officers_individual) {
```
Comment (AndreKramer7654) [inline test/utils/update/fetch.managing.officers.private.data.spec.ts:122]: would be better to not mutate appData mock for test order / concurrency

Comment (AndreKramer7654) [reply test/utils/update/fetch.managing.officers.private.data.spec.ts:122]: This case should really never happen - the review list MOs should always have ch-references. Same for line 136 below.

Comment (bmcgreevyCH) [reply test/utils/update/fetch.managing.officers.private.data.spec.ts:122]: Removed

**Review by AndreKramer7654 — APPROVED** [review]

**Review by azafarCH — APPROVED** [review]

**Pull request merged**